invoices/invoice_expiry_watcher.go

```diff
@@ -128,48 +129,23 @@ func (ew *InvoiceExpiryWatcher) prepareInvoice(
 }
 
 // AddInvoices adds multiple invoices to the InvoiceExpiryWatcher.
-func (ew *InvoiceExpiryWatcher) AddInvoices(
-	invoices map[lntypes.Hash]*channeldb.Invoice) {
-
-	invoicesWithExpiry := make([]*invoiceExpiry, 0, len(invoices))
-	for paymentHash, invoice := range invoices {
-		newInvoiceExpiry := ew.prepareInvoice(paymentHash, invoice)
-		if newInvoiceExpiry != nil {
-			invoicesWithExpiry = append(
-				invoicesWithExpiry, newInvoiceExpiry,
-			)
-		}
-	}
-
-	if len(invoicesWithExpiry) > 0 {
+func (ew *InvoiceExpiryWatcher) AddInvoices(invoices []*invoiceExpiry) {
+	if len(invoices) > 0 {
 		log.Debugf("Added %d invoices to the expiry watcher",
-			len(invoicesWithExpiry))
+			len(invoices))
 		select {
-		case ew.newInvoices <- invoicesWithExpiry:
+		case ew.newInvoices <- invoices:
 		// Select on quit too so that callers won't get blocked in case
 		// of concurrent shutdown.
 		case <-ew.quit:
 		}
 	}
 }
 
-// AddInvoice adds a new invoice to the InvoiceExpiryWatcher. This won't check
-// if the invoice is already added and will only add invoices with ContractOpen
-// state.
-func (ew *InvoiceExpiryWatcher) AddInvoice(
-	paymentHash lntypes.Hash, invoice *channeldb.Invoice) {
-
-	newInvoiceExpiry := ew.prepareInvoice(paymentHash, invoice)
-	if newInvoiceExpiry != nil {
-		log.Debugf("Adding invoice '%v' to expiry watcher,"+
-			"expiration: %v", paymentHash, newInvoiceExpiry.Expiry)
-
-		select {
-		case ew.newInvoices <- []*invoiceExpiry{newInvoiceExpiry}:
-		// Select on quit too so that callers won't get blocked in case
-		// of concurrent shutdown.
-		case <-ew.quit:
-		}
+// AddInvoice adds a new invoice to the InvoiceExpiryWatcher.
```

Comment:
nit: godoc here could explain that nil input is a nop

Reply:
So now, even if we were to add a `nil` entry, it'll be discarded in the `select` (below).